src/components/smart_objects/src/object_schema_item.cc

```diff
@@ -78,29 +88,46 @@ SMember::SMember(const ISchemaItemPtr SchemaItem,
 bool SMember::CheckHistoryFieldVersion(
     const utils::SemanticVersion& MessageVersion) const {
   if (MessageVersion.isValid()) {
-    if (mSince != boost::none) {
-      if (MessageVersion < mSince.get()) {
-        return false;  // Msg version predates `since` field
-      } else {
-        if (mUntil != boost::none && (MessageVersion >= mUntil.get())) {
-          return false;  // Msg version newer than `until` field
-        } else {
-          return true;  // Mobile msg version falls within specified version
-                        // range
-        }
-      }
-    }
-
-    if (mUntil != boost::none && (MessageVersion >= mUntil.get())) {
-      return false;  // Msg version newer than `until` field
+    if (mType == Type::OEM_SPECIFIC) {
+      return CheckCustomVehicleData(MessageVersion);
     } else {
-      return true;  // Mobile msg version falls within specified version range
+      return CheckAPIVehicleData(MessageVersion);
     }
   }
 
   return true;  // Not enough version information. Default true.
 }
 
+bool SMember::CheckCustomVehicleData(
+    const utils::SemanticVersion& MessageVersion) const {
+  if (mSince != boost::none) {
+    if (MessageVersion < mSince.get() && MessageVersion < kModuleVersion) {
```

Comment:
Instead of treating policy types differently, I think it would be better to just keep any members which have a `since` value above the module version, since such a value would not exist in the API. These functions and the `Type` enum shouldn't be necessary in that case.

Reply:
@jacobkeeler please take a look at 986bcf07dafae78bff53a2be8a14998b251ef955